Functions/xASL_adm_LoadX.m

```diff
@@ -65,21 +65,40 @@
 end
 
 %% 3. Look for and update deprecated fields
-nameConversionTable = xASL_adm_LoadX_DeprecatedNamesTable();
+conversionTable = xASL_adm_LoadX_DeprecatedNamesTable();
 
+nOutdatedParameter = 0;
 bOldFieldsDetected = 0;
-for iField = 1:size(nameConversionTable,1)
-	if isfield(OldX.x,nameConversionTable{iField,1})
-		bOldFieldsDetected = 1;
-		if ~isfield(OldX.x,nameConversionTable{iField,2}) || ~isfield(OldX.x.(nameConversionTable{iField,2}),nameConversionTable{iField,3})
-			OldX.x.(nameConversionTable{iField,2}).(nameConversionTable{iField,3}) = OldX.x.(nameConversionTable{iField,1});
-		end
-		OldX.x = rmfield(OldX.x,nameConversionTable{iField,1});
-	end
+for iField = 1:size(conversionTable,1)
+    if isfield(OldX.x,conversionTable{iField,1})
+        bOldFieldsDetected = 1;
+        if ~isfield(OldX.x,conversionTable{iField,2}) || ...
+                ~isfield(OldX.x.(conversionTable{iField,2}),conversionTable{iField,3}) || ...
+                ~isfield(OldX.x.(conversionTable{iField,2}),conversionTable{iField,4})
+            % Deprecated field detected
+            nOutdatedParameter = nOutdatedParameter+1;
+            detectedFields{nOutdatedParameter,1} = conversionTable{iField,1};
+            if ~isfield(OldX.x.(conversionTable{iField,2}),conversionTable{iField,3})
+                % Sub-Structure (x.FIELD -> x.STRUCT.FIELD)
+                OldX.x.(conversionTable{iField,2}).(conversionTable{iField,3}) = OldX.x.(conversionTable{iField,1});
+            else
+                % Sub-Sub-Structure (x.FIELD -> x.STRUCT.STRUCT.FIELD)
+                OldX.x.(conversionTable{iField,2}).(conversionTable{iField,3}).(conversionTable{iField,4}) = OldX.x.(conversionTable{iField,1});
+            end
+        end
+        % Remove old field
+        OldX.x = rmfield(OldX.x,conversionTable{iField,1});
+    end
 end
 
 if bOldFieldsDetected
-	warning('Detected deprecated fields in x.mat. Older version of ExploreASL was used to process this data previously. Field names fixed');
+    % Write fields back
+    x = OldX.x;
```

Comment:
@jan-petr: Why didn't we write the OldX.x to x before? Otherwise all of this doesn't work at all 😄

Reply:
:D